fedimint-core/src/query.rs

```diff
@@ -62,11 +62,38 @@ impl<R: Debug + Eq + Clone> QueryStrategy<R> for VerifiableResponse<R> {
     }
 }
 
+struct ErrorStrategy {
+    errors: BTreeMap<PeerId, MemberError>,
+    threshold: usize,
+}
+
+impl ErrorStrategy {
+    pub fn new(threshold: usize) -> Self {
+        Self {
+            errors: BTreeMap::new(),
+            threshold,
+        }
+    }
+
+    pub fn process<R>(&mut self, peer: PeerId, error: MemberError) -> QueryStep<R> {
+        assert!(self.errors.insert(peer, error).is_none());
+
+        if self.errors.len() == self.threshold {
```

Comment:
If `threshold == 0` this could result in an endless loop, should the constructor just assert `threshold > 0` since 0 is nonsensical anyway?

Reply:
Yes